transports/tcp/src/lib.rs

```diff
@@ -108,7 +110,24 @@ impl Transport for $tcp_config {
         async fn do_listen(cfg: $tcp_config, socket_addr: SocketAddr)
             -> Result<impl Stream<Item = Result<ListenerEvent<Ready<Result<$tcp_trans_stream, io::Error>>, io::Error>, io::Error>>, io::Error>
         {
-            let listener = <$tcp_listener>::bind(&socket_addr).await?;
+            let socket = if socket_addr.is_ipv4() {
+                Socket::new(Domain::ipv4(), Type::stream(), Some(socket2::Protocol::tcp()))?
+            } else {
+                let s = Socket::new(Domain::ipv6(), Type::stream(), Some(socket2::Protocol::tcp()))?;
+                s.set_only_v6(true)?;
+                s
+            };
+            if cfg!(target_family = "unix") {
+                socket.set_reuse_address(true)?;
```

Comment:
I have no idea of the implications of that change. What are the reasons for that?

Reply:
I do not know why it is restricted to Unix, but again this is what MIO does which in turn does [what libstd does](https://github.com/rust-lang/rust/blob/c06e4aca19046b07d952b16e9f002bfab38fde6b/src/libstd/sys_common/net.rs#L393-L398).